### PR TITLE
chore(expandablesection): Added a demo to show how to use expandable section to create an expandable text component.

### LINKED ISSUE #9888: Create code sand box demo for Expandable Text
Based on PR patternfly/react-component-groups#39 the team has decided to do a demo instead of creating the Expandable Text component.  A code sandbox demo should be created that uses PF components to have the functionality as the Expandable Text component.

### PR DESCRIPTION
**Pull request by dlabaj**
Added a demo to show how to use expandable section to create an expandable text component.

Closes #9888

### CODE REVIEW

File: packages/react-core/src/demos/ExpandableSection/ExpandableSection.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+id: Expandable section
+section: components
+---
+
+## Demos
+
+- The following demo shows how to use the ExpandableSection to create an expandable text component.
+
+### Expandable Text Component
+
```
Comment (edonehoo) [inline packages/react-core/src/demos/ExpandableSection/ExpandableSection.md:11]:
```suggestion
### Expandable text with truncation

You can truncate long sections of text and add a "Show more" link that allows users to expand and view the view text content.

To specify the number of character shown prior to truncation, use a `<Truncate>` component and pass a `maxWidth` to the `style` attribute.

```

**Review by fhlavac — APPROVED** [review]
Looks good!

**Review by thatblindgeye — COMMENTED** [review]
One non-blocking nit I have is that we could update the demo name to "Expandable text with character truncation" to more immediately differentiate it from the existing truncate expansion, but wouldn't hold this up over that.

Otherwise only other thing is that Truncate adds a tooltip to the content, which we don't really need since we have the toggle to show the rest of the content. That could also be an issue if the truncated content is large since it'll create an equally large tooltip that would either just be a distraction or be cut off. Do we need to use a Truncate since this is more a demo showcasing how to achieve this result? I was able to get a similar (though possibly not perfect) result by using a div instead:

`<div style={{ maxWidth: '10ch' }}>{content}</div>`

We could also allow preventing a tooltip from with Truncate. I'd be hesitant about that being overridden since typically we should want that tooltip for truncated content, but a good prop description + a11y docs might offset that.

**Review by andrew-ronaldson — APPROVED** [review]
LGTM, thanks!

File: packages/react-core/src/demos/ExpandableSection/ExpandableSection.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+id: Expandable section
+section: components
+---
+
+## Demos
+
+
+### Expandable text with truncation
```
Comment (edonehoo) [inline packages/react-core/src/demos/ExpandableSection/ExpandableSection.md:9]:
```suggestion
### Expandable text with character truncation
```

**Review by edonehoo — CHANGES_REQUESTED** [review]
small change per Eric's suggestion, which I +1

**Review by evwilkin — APPROVED** [review]
LGTM

With regard to @thatblindgeye comment about wrapping with a div vs using Truncate, I get the entire text block limited width as seen below - if any follow-up is needed we can open a new issue but IMO this is good to go as-is for now

https://github.com/patternfly/patternfly-react/assets/8651509/906cb2f7-c073-4338-b039-f536d737a489

**Review by kmcfaul — APPROVED** [review]

**Review by mcoker — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]
LGTM.  @dlabaj is out.  We can talk to him about your suggestion and follow up if needed once he returns

**Pull request merged**